backend/algorithms/objects/conditions.py

```diff
@@ -1,75 +1,69 @@
 '''
 Contains the Conditions classes
 '''
+import abc
 import json
 from algorithms.objects.helper import is_course, is_program
 from algorithms.objects.categories import AnyCategory
 
-
 '''Keywords'''
 AND = 1
 OR = 2
 
 '''CACHED'''
-# Load in cached exclusions
-CACHED_EXCLUSIONS_PATH = "./algorithms/cache/exclusions.json"
-with open(CACHED_EXCLUSIONS_PATH) as f:
-    CACHED_EXCLUSIONS = json.load(f)
-    f.close()
-
-CACHED_CONDITIONS_TOKENS_PATH = "./data/finalData/conditionsTokens.json"
+CACHED_CONDITIONS_TOKENS_PATH = "./data/final_data/conditionsTokens.json"
 with open(CACHED_CONDITIONS_TOKENS_PATH) as f:
     CACHED_CONDITIONS_TOKENS = json.load(f)
-    f.close()
 
 
 CACHED_PRGORAM_MAPPINGS_FILE = "./algorithms/cache/programMappings.json"
 with open(CACHED_PRGORAM_MAPPINGS_FILE) as f:
     CACHED_PRGORAM_MAPPINGS = json.load(f)
-    f.close()
 
-class CourseCondition():
+class Condition():
+    @abc.abstractmethod
+    def validate(self, user) -> tuple[bool, list[str]]:
+        pass
```

Comment:
maybe add a line of comment here so that other people know what the returned values mean. The tuple is not self-explanatory